src/mongocxx/read_concern.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 MongoDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <mongocxx/read_concern.hpp>
+#include <mongocxx/private/read_concern.hpp>
+
+#include <bsoncxx/stdx/make_unique.hpp>
+#include <mongocxx/private/libmongoc.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+
+read_concern::read_concern() : _impl{stdx::make_unique<impl>(libmongoc::read_concern_new())} {
+}
+
+read_concern::read_concern(std::unique_ptr<impl>&& implementation) {
+    _impl.reset(implementation.release());
```

Comment:
does

`
_impl = std::move(implementation)
`

work? That seems more idiomatic.